protocol/src/main/java/org/sdo/pri/OnDieSignatureValidator.java

```diff
@@ -0,0 +1,197 @@
+   * @return boolean indicating if signature is valid.
+   * @throws CertificateException when error.
+   */
+  public static boolean validateWithRevocations(
+          String signedData,
+          byte[] signature,
+          CertPath certPath,
+          OnDieCache onDieCache) throws CertificateException {
+
+    // Check revocations first.
+    List<Certificate> certificateList = (List<Certificate>) certPath.getCertificates();
+
+    // Check revocations first.
+    if (!checkRevocations(certificateList, onDieCache)) {
+      return false;
+    }
+
+    // Now validate the signature
+    // Public key comes from the cert chain in the voucher.
+    if (certificateList.size() == 0) {
+      throw new CertificateException("OnDieSignatureValidation: Certificate chain is empty.");
+    }
+    return validateWithoutRevocations(signedData, signature, certificateList.get(0).getPublicKey());
+  }
+
+
+  private static byte[] convertSignature(byte[] signature, byte[] taskInfo)
+        throws IllegalArgumentException, IOException {
+    if (taskInfo.length != 36) {
+      throw new IllegalArgumentException("taskinfo length is incorrect: " + taskInfo.length);
+    }
+
+    // Format for signature should be as follows:
+    // 0x30 b1 0x02 b2 (vr) 0x02 b3 (vs)
+    // The b1 = length of remaining bytes,
+    // b2 = length of R value (vr), b3 = length of S value (vs)
+    byte[] rvalue = Arrays.copyOfRange(signature, taskInfo.length, taskInfo.length + 48);
+    byte[] svalue = Arrays.copyOfRange(signature, taskInfo.length + 48, taskInfo.length + 96);
+
+    // format signature: if upper most bit is set then prepend with a 0x00 and increase length by 1
+    boolean appendZeroToR = false;
+    boolean appendZeroToS = false;
+    if ((rvalue[0] & 0x80) != 0) {
+      appendZeroToR = true;
+    }
+    if ((svalue[0] & 0x80) != 0) {
+      appendZeroToS = true;
+    }
+
+    ByteArrayOutputStream adjSignature = new ByteArrayOutputStream();
+    adjSignature.write(0x30);
+    // total length of remaining bytes
+    adjSignature.write(4 + (appendZeroToR ? 49 : 48) + (appendZeroToS ? 49 : 48));
+    adjSignature.write(0x02);
+    // R value
+    if (appendZeroToR) {
+      adjSignature.write(49);
+      adjSignature.write(0x00);
+      adjSignature.write(rvalue);
+    } else {
+      adjSignature.write(48);
+      adjSignature.write(rvalue);
+    }
+    adjSignature.write(0x02);
+    // S value
+    if (appendZeroToS) {
+      adjSignature.write(49);
+      adjSignature.write(0x00);
+      adjSignature.write(svalue);
+    } else {
+      adjSignature.write(48);
+      adjSignature.write(svalue);
+    }
+    return adjSignature.toByteArray();
+  }
+
+
+  private static boolean checkRevocations(
+          List<Certificate> certificateList,
+          OnDieCache onDieCache) {
+    // Check revocations first.
+    try {
+      CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+      for (Certificate cert: certificateList) {
```

Comment:
Follow one style for iterator .. preferably (Certificate cert : certificateList) .. same for other instance

I suppose, this should have been caught by the checkstyle, but somehow is not checked.

Reply:
done